src/pages/Touringcar/contexts/MapProvider.tsx

```diff
@@ -1,20 +1,46 @@
-import { ReactNode, useReducer, useState } from 'react'
+import { ReactNode, useCallback, useReducer, useState } from 'react'
 
 import { TouringcarParkingSpace } from 'api/touringcar/parking-spaces'
+import { useSearchParams } from 'react-router-dom'
 
-import { MapPanelTab, TouringcarMapContext } from './MapContext'
+import {
+  MapLayerParam,
+  MapLayerParamToMapLayer,
+  MapPanelTab,
+  TouringcarMapContext,
+  mapLayerParamIds,
+} from './MapContext'
 import mapLayersReducer, { mapLayersInitialState } from './mapLayersReducer'
 
-type Props = {
-  children: ReactNode
-}
-
-export const TouringcarMapProvider = ({ children }: Props) => {
+function TouringcarMapProvider({ children }: { children: ReactNode }) {
   const [activeMapLayers, updateActiveMapLayers] = useReducer(
     mapLayersReducer,
     mapLayersInitialState
   )
 
+  const [queryParams] = useSearchParams()
+  const updateActiveMapLayersWithSearchParams = useCallback(() => {
+    const mapLayerParams: MapLayerParam[] = (
+      [...queryParams.keys()] as MapLayerParam[]
+    ).filter(key => {
+      return mapLayerParamIds.includes(key as MapLayerParam)
+    })
+
+    mapLayerParamIds.forEach(param => {
+      const queryParam = MapLayerParamToMapLayer[
+        param
+      ] as unknown as keyof typeof mapLayersInitialState
+
+      if (!mapLayerParams.length) return
```

Comment:
Can we move this if statement outside the loop? I think if it doesn't execute once it will never execute within this loop right? So might as well not run the loop at all.

Reply:
The var naming is confusing, but this one is different from `mapLayersParamIds` that we are looping over. If there are no relevant URL parameters, we want to escape the function before legend items (un)checked.

Reply:
Sorry, didn't get your request. Yes, moving it outside the loop.